src/Microsoft.AspNetCore.Rewrite/UrlRewriteExtensions.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Rewrite;
+
+namespace Microsoft.AspNetCore.Builder
+{
+    /// <summary>
+    /// Extension methods for the <see cref="UrlRewriteMiddleware"/> 
+    /// </summary>
+    public static class UrlRewriteExtensions
+    {
+        /// <summary>
+        /// Checks if a given Url matches rules and conditions, and modifies the HttpContext on match.
+        /// </summary>
+        /// <param name="app"></param>
+        /// <param name="action">Action to apply to the <see cref="UrlRewriteOptions"/> </param>
+        /// <returns></returns>
+        public static IApplicationBuilder UseRewriter(this IApplicationBuilder app, Action<UrlRewriteOptions> action)
```

Comment:
This doesn't quite follow our other middleware patterns. We moved away from the lambda pattern and just pass Options directly (or nothing).